repository/src/main/java/org/alfresco/repo/event2/EventGenerator.java

```diff
@@ -311,12 +311,24 @@ public void onRemoveAspect(NodeRef nodeRef, QName aspectTypeQName)
     public void onCreateChildAssociation(ChildAssociationRef childAssociationRef, boolean isNewNode)
     {
         getEventConsolidator(childAssociationRef).onCreateChildAssociation(childAssociationRef, isNewNode);
+        if (!childAssociationRef.isPrimary())
+        {
+            // if this is a secondary relationship simulate node move event to store state of previous secondary parents
+            ChildAssociationRef oldChildAssociationRef = childAssociationWithoutParentOf(childAssociationRef);
```

Comment:
The secondary parent may already be present with a different association type. For example we can create secondary child associations with this request body:
```
[
  {
    "childId": "7bb9c846-fcc5-43b5-a893-39e46ebe94d4",
    "assocType": "cm:preferenceImage"
  },
  {
    "childId": "7bb9c846-fcc5-43b5-a893-39e46ebe94d4",
    "assocType": "cm:contains"
  }
]
```

Reply:
I have raised ACS-6097 for this edge case so that we can continue with this PR.